src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs

```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
+        /// <param name="strides">The strides of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
         static abstract TSelf CreateUninitialized(scoped ReadOnlySpan<nint> lengths, scoped ReadOnlySpan<nint> strides, bool pinned = false);
 
+        /// <summary>
+        /// Gets whether the collection is read-only.
+        /// </summary>
         bool IsReadOnly { get; }
 
+        /// <summary>
+        /// Gets the value at the specified indexes.
+        /// </summary>
+        /// <param name="indexes">The indexes to use.</param>
         new T this[params scoped ReadOnlySpan<nint> indexes] { get; set; }
+
+        /// <summary>
+        /// Gets the value at the specified indexes.
+        /// </summary>
+        /// <param name="indexes">The indexes to use.</param>
         new T this[params scoped ReadOnlySpan<NIndex> indexes] { get; set; }
+
+        /// <summary>
+        /// Gets the values at the specified ranges.
+        /// </summary>
+        /// <param name="ranges">The ranges to be used.</param>
         new TSelf this[params scoped ReadOnlySpan<NRange> ranges] { get; set; }
 
+        /// <summary>
+        /// Creates a tensor span for the entire underlying buffer.
+        /// </summary>
+        /// <returns>The converted <see cref="TensorSpan{T}"/>.</returns>
         TensorSpan<T> AsTensorSpan();
+
+        /// <summary>
+        /// Creates a tensor span for the specified start indexes.
+        /// </summary>
+        /// <param name="start">The start locations to be used.</param>
+        /// <returns>The converted <see cref="TensorSpan{T}"/>.</returns>
         TensorSpan<T> AsTensorSpan(params scoped ReadOnlySpan<nint> start);
+
+        /// <summary>
+        /// Creates a tensor span for the specified start indexes.
+        /// </summary>
+        /// <param name="startIndex">The start indexes to be used.</param>
+        /// <returns>The converted <see cref="TensorSpan{T}"/>.</returns>
         TensorSpan<T> AsTensorSpan(params scoped ReadOnlySpan<NIndex> startIndex);
+
+        /// <summary>
+        /// Creates a tensor span for the specified ranges.
+        /// </summary>
+        /// <param name="range">The ranges to be used.</param>
+        /// <returns>The converted <see cref="TensorSpan{T}"/>.</returns>
         TensorSpan<T> AsTensorSpan(params scoped ReadOnlySpan<NRange> range);
 
+        /// <summary>
+        /// Clears the tensor.
+        /// </summary>
         void Clear();
+
+        /// <summary>
+        /// Fills the contents of this tensor with the given value.
+        /// </summary>
         void Fill(T value);
+
+        /// <summary>
+        /// Returns a reference to the 0th element of the tensor. If the tensor is empty, returns null reference.
+        /// It can be used for pinning and is required to support the use of the tensor within a fixed statement.
```

Comment:
```suggestion
        /// This method can be used for pinning and is required to support the use of the tensor within a fixed statement.
```

Reply:
Also, this sentence should go in remarks, not summary.

Reply:
Fixed.